events/importer/helper/importers/vapaaehtoistyofi/reader.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import logging
+from .record import Record
+
+log = logging.getLogger(__name__)
+
+
+class Reader:
+    endpoint_url = 'https://apiv2.vapaaehtoistyo.fi'
+    rest_user_agent = 'HelsinkiVETImporter/0.1'
+    timeout = 5.0
+    cached_entries = True
+
+    def __init__(self, api_key):
+        if not api_key:
+            raise ValueError("Really need API-key!")
+        self.api_key = api_key
+
+        if self.cached_entries:
+            cnt_entries, data = self.load_entries()
+            self.entries = {}
+            for record in data:
+                id = record.id
+                self.entries[id] = record
+
+    def _setup_client(self):
+        headers = {
+            'Accept': 'application/json',
+            'User-Agent': self.rest_user_agent,
+            "Authorization": "Bearer %s" % self.api_key
+        }
+
+        s = requests.Session()
+        s.headers.update(headers)
+
+        return s
+
+    def load_entry(self, id):
+        if not self.cached_entries:
+            return self._load_entry_api(id)
+
+        if id not in self.entries:
+            return False
+
+        return self.entries[id]
+
+    def _load_entry_api(self, id):
+        http_client = self._setup_client()
+        url = "%s/task/%s" % (self.endpoint_url, id)
+        response = http_client.get(url, timeout=self.timeout)
+        if response.status_code != 200:
+            raise RuntimeError("Failed to request data from Vapaaehtoistyö.fi API! HTTP/%d" %
+                               response.status_code)
+
+        data = response.json()
+        if 'status' not in data or data['status'] != "ok":
+            raise RuntimeError("Vapaaehtoistyö.fi response isn't ok!")
+        if 'data' not in data:
+            raise RuntimeError("Vapaaehtoistyö.fi response doesn't contain 'data'!")
+        data_obj = Record(data)
+
+        return data_obj
+
+    def load_entries(self):
+        http_client = self._setup_client()
+        page = 1
+        total_records = None
+        batch_size = 1
```

Comment:
I'd get rid of `batch_size` here, change the while loop to `True`, then break the loop if `len(data['data']['records']) == 0`. This could make this shorter and a bit easier to understand

Reply:
This suggestion is heavily opinionated.
I typically tend not to use `while True` -stucture as it can yield a forever-loop.